src/mbgl/tile/vector_tile_data.cpp

```diff
@@ -77,6 +79,11 @@ VectorTileData::VectorTileData(const TileID& id_,
                 // existing buckets in case we got a refresh parse.
                 buckets = std::move(resultBuckets.buckets);
 
+                if (state == State::parsed) {
```

Comment:
I'm not 100% sure I understand how partial tile parsing works, but:
- `parseGeometryTile` [calls](https://github.com/mapbox/mapbox-gl-native/blob/7fa32a0d083b9c828d2e487f547d55967bd43817/src/mbgl/util/worker.cpp#L37) `parseAllLayers`
- `parseAllLayers` only returns `parsed` if it [manages to parse all layers](https://github.com/mapbox/mapbox-gl-native/blob/7fa32a0d083b9c828d2e487f547d55967bd43817/src/mbgl/tile/tile_worker.cpp#L62) which I think happens when it doesn't need to load any glyphs or icons